### PR TITLE
Change TensorExpressions interface via changing namespaces

### PR DESCRIPTION
**Pull request by macedo22**
## Proposed changes

This PR changes two things about the `TensorExpression` interface:
- `namespace TensorExpressions` -> `namespace tenex` to reduce keystrokes for calling `TensorExpressions::evaluate` and `TensorExpressions::update`
- Generic tensor indices `ti_a`, etc. are now in their own `ti` namespace and are accessed via `ti::a`, etc. to make it more consistent with our `tnsr::ij`, etc. aliases

### Upgrade instructions

- Anywhere using `TensorExpressions` namespace should now use `tenex`
- Anywhere using `ti_a` (tensor indices) should now use `ti::a`

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/DataStructures/Tensor/Expressions/AddSubtract.hpp [hunk 1]
```diff
@@ -28,17 +28,17 @@
 #include "Utilities/TMPL.hpp"
 
 /// \cond
-namespace TensorExpressions {
+namespace tenex {
```
Comment (nilsvu) [inline src/DataStructures/Tensor/Expressions/AddSubtract.hpp:31]: How about placing these things directly in `tnsr` so we don't have to remember another namespace?

**Review by nilsvu — APPROVED** [review]

File: src/DataStructures/Tensor/Expressions/AddSubtract.hpp (hunk 1 above)
Comment (macedo22) [reply src/DataStructures/Tensor/Expressions/AddSubtract.hpp:31]: Hmmm this is just my opinion, but I think that might make things a little bit confusing because I already have it in my mind that `tnsr`s are tensors, but the TE generic indices are a different kind of thing (not tensors).

In addition, I think the naming convention for either the tensor aliases or the TE generic tensor indices would have to change because we would have a naming conflict with these: `tnsr::i`, `tnsr::I`, `tnsr::a`, and `tnsr::A`.

How do you feel about these reasons?

Comment (macedo22) [reply src/DataStructures/Tensor/Expressions/AddSubtract.hpp:31]: P.S. and `tnsr` more keystrokes than `ti`

Comment (macedo22) [reply src/DataStructures/Tensor/Expressions/AddSubtract.hpp:31]: P.S. and `tnsr` is more keystrokes than `ti`

Comment (nilsvu) [reply src/DataStructures/Tensor/Expressions/AddSubtract.hpp:31]: Yes I think `ti` is great for the tensor indices, it's very short, clear, and mirrors the `tnsr::i` aliases.

My question relates only to the first commit that changes `TensorExpressions -> tenex`. Should we do `TensorExpressions -> tnsr` instead? This sort of "promotes" the tensor expressions into the tensor namespace and reads `tnsr::evaluate()`. What do you think?

Comment (macedo22) [reply src/DataStructures/Tensor/Expressions/AddSubtract.hpp:31]: Whoops, I spaced and missed where your review comment was, sorry.

TLDR I don't have strong feelings

Longer version:

I don't think I'm particularly for or against it strongly, but I would imagine some people may be opposed for a similar reason to what I gave above: that people may like having `tnsr` specific to the tensor aliases, in the sense that by having `tnsr::evaluate` (and all of the internal TE stuff that's not user facing also be in this namespace) makes things less clean and more heterogeneous? Also, the `tnsr::` part of `tnsr::ij` feels like it's part of the name of the type or something, meaning that it feels like it specifically refers to a tensor type (not other things). However, I guess this is mostly just a consequence of our current usage of `tnsr` and it doesn't *have* to be specifically tensor aliases only, especially if we think of TEs as just a part of the `Tensor` interface/package. Plus, it's possible to put just `evaluate` and `update` (the only user-facing things) in the `tnsr` namespace but leave all of the internal stuff in `tenex`/`TensorExpressions`. I don't know why I didn't consider that until now.

I'm not making a very strong case either way. I kind of like having the tensor aliases having their own designation, like a namespace specifically for tensor type shorthands, but I am not strongly opposed to lumping TE stuff in with it. If we do want to do `tnsr::evaluate` and `tnsr::update`, then I would advocate for leaving all non-user-facing internal TE stuff to remain in its own namespace, just because it doesn't feel clean to have all of the TE internals in `tnsr` for some reason.

I believe @nilsdeppe wasn't a fan of putting TE stuff in the `tnsr` namespace, but I don't recall why. Maybe he will have a more useful opinion than I've presented.

Comment (nilsvu) [reply src/DataStructures/Tensor/Expressions/AddSubtract.hpp:31]: Either way works for me

Comment (macedo22) [reply src/DataStructures/Tensor/Expressions/AddSubtract.hpp:31]: Okay, I guess we can wait for Nils Deppe because I think he might have a horse in this race

Comment (nilsdeppe) [reply src/DataStructures/Tensor/Expressions/AddSubtract.hpp:31]: `tnsr` was designated for shortcuts to longer `Tensor<...>`, similar to `ti` for the indices. Since the `Tensor` class doesn't live in the `tnsr` namespace, I see no reason to put the expressions in there.

Comment (macedo22) [reply src/DataStructures/Tensor/Expressions/AddSubtract.hpp:31]: Ok, we can leave it as is

**Review by nilsdeppe — APPROVED** [review]

**Comment by nilsdeppe** [comment]
Merging now since I don't want this to collect git conflicts...
